### PR TITLE
Improve flickr code quality

### PR DESCRIPTION
**Pull request by abdala2020**
To improve the code quality, here are the changes I have made to 
the flickr.py 
1-	Image_data(): is a function that returns data or no data if the connection is successful and different error msg for different response errors
2-	Extract_img_url_params(): this function uses the image_data and extracts the params needs to get the drink image 
3-	Retrieve_image(): uses the params extracted and finds a drink image and returns the response 
4-	Create_img_file(): this creates a file for the image in the images folder. And returns the file name
I also exclude the image file from the git repo

Changes made to api_controller.py:
In order not to mess up with the work that has already been done. I created the following function.
1-	Handle_flickr_api_response(): this function handles the API response which could be (data, no data, error response – either error in querying or connection error)
2-	Then I replaced the getImage() in the get_food_info() with the function that handles the Flickr API. 

finally, I created a test file for flicker, and so far I was only able to test only one scenario.

### CODE REVIEW

**Comment by 7ibakana** [comment]
lots of good work

File: api/flickr.py [hunk 1]
```diff
@@ -1,31 +1,61 @@
...
+def get_image_data(drinkName):
+    """Set up the parameters required in the search"""
+    param = {
+    'method' : 'flickr.photos.search',
+    'text' : drinkName,
+    'media': 'photos',
+    'per_page' : 1,
+    'sort' : 'relevance',
+    'format': 'json',
+    'nojsoncallback' : '1n',
```
Comment (sunnycake) [inline api/flickr.py]: @abdala2020 I think you might've added the 'n' here accidently?

Comment (sunnycake) [reply api/flickr.py]: I took it out.

**Comment by sunnycake** [comment]
resolved conflict.

**Pull request merged**